graph/memstore/quadstore.go

```diff
@@ -193,7 +193,7 @@ func (qs *QuadStore) indexOf(t quad.Quad) (int64, bool) {
 
 func (qs *QuadStore) AddDelta(d graph.Delta) error {
 	if _, exists := qs.indexOf(d.Quad); exists {
-		return graph.ErrQuadExists
+		return &graph.DeltaError{Delta: d, Err: graph.ErrQuadExists}
```

Comment:
QuadWriter interface, but yes, this gets called by ApplyDeltas. I think it's a relic, rather than an addition, the QuadWriter approach came after the simple Quad addition.